pip/basecommand.py

```diff
@@ -104,6 +114,9 @@ def parse_args(self, args):
         # factored out for testability
         return self.parser.parse_args(args)
 
+    def running_outside_virtualenv(self):
```

Comment:
I don't think that's too bad. If we had it as `.in_virtualenv()` then it would look like:

```python
if not self.in_virtualenv():
    ...
```

I actually don't like function that's have negative logic because if I want to reverse the logic, then I have to read a double-negative logic and my brain starts to melt. I'd rather read positive logic and then just add a `not` to negate it.